barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_ultra_circuit_builder.cpp

```diff
@@ -0,0 +1,165 @@
+ *
+ * @param op_idx Index of op code in variables array
+ * @param point
+ * @param scalar
+ * @return ecc_op_tuple Tuple of indices into variables array used to construct pair of ecc op gates
+ */
+template <typename FF>
+ecc_op_tuple GoblinUltraCircuitBuilder_<FF>::decompose_ecc_operands(uint32_t op_idx,
+                                                                    const g1::affine_element& point,
+                                                                    const FF& scalar)
+{
+    // Decompose point coordinates (Fq) into hi-lo chunks (Fr)
+    const size_t CHUNK_SIZE = 2 * DEFAULT_NON_NATIVE_FIELD_LIMB_BITS;
+    auto x_256 = uint256_t(point.x);
+    auto y_256 = uint256_t(point.y);
+    auto x_lo = FF(x_256.slice(0, CHUNK_SIZE));
+    auto x_hi = FF(x_256.slice(CHUNK_SIZE, CHUNK_SIZE * 2));
+    auto y_lo = FF(y_256.slice(0, CHUNK_SIZE));
+    auto y_hi = FF(y_256.slice(CHUNK_SIZE, CHUNK_SIZE * 2));
+
+    // Split scalar into 128 bit endomorphism scalars
+    FF z_1 = 0;
+    FF z_2 = 0;
+    auto converted = scalar.from_montgomery_form();
+    FF::split_into_endomorphism_scalars(converted, z_1, z_2);
+    z_1 = z_1.to_montgomery_form();
+    z_2 = z_2.to_montgomery_form();
+
+    // Populate ultra ops in OpQueue with the decomposed operands
+    op_queue->ultra_ops[0].emplace_back(this->variables[op_idx]);
+    op_queue->ultra_ops[1].emplace_back(x_lo);
+    op_queue->ultra_ops[2].emplace_back(x_hi);
+    op_queue->ultra_ops[3].emplace_back(y_lo);
+
+    op_queue->ultra_ops[0].emplace_back(this->variables[op_idx]);
+    op_queue->ultra_ops[1].emplace_back(y_hi);
+    op_queue->ultra_ops[2].emplace_back(z_1);
+    op_queue->ultra_ops[3].emplace_back(z_2);
+
+    // Add variables for decomposition and get indices needed for op wires
+    auto x_lo_idx = this->add_variable(x_lo);
+    auto x_hi_idx = this->add_variable(x_hi);
+    auto y_lo_idx = this->add_variable(y_lo);
+    auto y_hi_idx = this->add_variable(y_hi);
+    auto z_1_idx = this->add_variable(z_1);
+    auto z_2_idx = this->add_variable(z_2);
+
+    return { op_idx, x_lo_idx, x_hi_idx, y_lo_idx, y_hi_idx, z_1_idx, z_2_idx };
+}
+
+/**
+ * @brief Add ecc operation to queue
+ *
+ * @param in Variables array indices corresponding to operation inputs
+ * @note We dont explicitly set values for the selectors here since their values are fully determined by
+ * num_ecc_op_gates. E.g. in the composer we can reconstruct q_ecc_op as the indicator on the first num_ecc_op_gates
+ * indices. All other selectors are simply 0 on this domain.
+ */
+template <typename FF> void GoblinUltraCircuitBuilder_<FF>::populate_ecc_op_wires(const ecc_op_tuple& in)
+{
+    ecc_op_wire_1.emplace_back(in.op);
+    ecc_op_wire_2.emplace_back(in.x_lo);
+    ecc_op_wire_3.emplace_back(in.x_hi);
+    ecc_op_wire_4.emplace_back(in.y_lo);
+
+    ecc_op_wire_1.emplace_back(in.op); // TODO(luke): second op val is sort of a dummy. use "op" again?
```

Comment:
Mb set to zero? Could you also add an issue number to the todo?

Reply:
Yeah I think 0 is the right thing so updated and removed the TODO